utils/errors.go

```diff
@@ -0,0 +1,35 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/rs/zerolog/log"
+)
+
+// FlagError is the kind of error raised in flag processing
+type FlagError struct {
+	Err error
+}
+
+func (fe FlagError) Error() string {
+	return fe.Err.Error()
+}
+
+func (fe FlagError) Unwrap() error {
+	return fe.Err
+}
+
+// ErrorSilent is an error that triggers exit code 1 without any error messaging
+var ErrorSilent = errors.New("SilentError")
+
+// PrintError Prints in specified format
+func PrintError(err error, errString string, hasDebug bool) {
+
+	if err == ErrorSilent {
+		return
+	}
+	if hasDebug {
+		panic(errString)
+	}
+	log.Fatal().Msgf(errString)
+}
```

Comment:
Bit hard to digest the purpose of this function. Error which we can't be handled should be under panic. You can control log level through logging module, not by passing boolean flag all over the function.

Reply:
Do we want to get rid of Panic..?

Reply:
No, panic should be there on default for the error cases where it can't be recovered.

Reply:
Generally, all errors are nonrecoverable.
Like "Invalid Synk Token", "Server Error" and so.
If you want to get rid of Panic from here...let me know..ofcause removing this will reduce a `debug` arg to this function.

My Idea for this Function is Not to split Error Handling all over the code. Let me know if this doesn't sound okay..

Reply:
You can have a common file to represent all types of errors which this cli will throw.

e.g
```
func UserAuthError(...) error {
    return errors.New("Unable to authenticate user %s", userName)
)

func InvalildHostError(...) error {
...
}

...
```

These error generating function shall be called from anywhere. This streamlines error generation.

Reply:
`PrintError(err, "Error String", debug)` is built to handle Error, which means it takes `err` as an Argument.

I can't understand, how 
```
UserAuthError -> Error
```
and 
```
InvalildHostError -> Error
```
are likely its replacements..?

Reply:
and we are using Server generated Error Message, I avoided generating any New Error Message in CLI itself.

Reply:
FIXED. replaced with `log.Fatal().Msgf()`